%\subsection{Functional Annotation of Maize Genes}
%Three methods were used to annotate GO terms to genes: sequence similarity, domain presence, and mixed-method pipelines (See Figure \ref{fig:methods}).
%Sequence similarity using RBH identified putative Arabidopsis orthologs and GO terms associated were inherited (from TAIR) The same methodology identified orthologous sequences in the ten plant genomes with the most GO terms annotated (from UniProt). %Using BLASTP, maize genes were compared to \emph{Arabidopsis thaliana} gene sequences from TAIR and plant protein sequences from UniProt. The Reciprocal-Best-Hit (RBH) method was used to determine putative ortholog pairs between maize and other datasets. GO terms were inherited from the other plant gene to the maize gene within each putative ortholog pair. At the end of this process, two GO annotation datasets from the sequence-similarity based methods are generated for maize, namely TAIR-maize GO dataset and UniProt-maize GO dataset.
%The InterProScan (IPRS) pipeline was chosen to annotate GO terms to maize genes using the domain presence method. Maize genes were annotated using IPRS with default parameters, and a single IPRS-maize GO dataset was produced as a result.
%IPRS pipeline detects presence of protein domains and annotated GO terms to protein sequences based on those domains. 
%Maize genes were also annotated by three different mixed-method pipelines related to the CAFA competition, namely Argot2, PANNZER, and FANN-GO. Maize sequences were preprocessed as required by specific pipelines and were annotated using each pipeline mentioned above.  This procedure resulted in 3 different raw datasets. The raw datasets from mixed-method pipelines were cleaned by removing low-confidence annotations.  This was performed by determining GO category specific annotation score thresholds which produced the highest Avg F-score independently for each raw dataset. The annotations with a score below the determined score threshold were filtered out to clean the raw datasets. The three resulting datasets from mixed-method pipelines are, Argot2-maize GO dataset, FANN-GO-maize GO dataset, and PANNZER-maize GO dataset.

%GO annotation of maize genes yielded six maize GO datasets produced by the different GO annotation methods used in the maize-GAMER. Each GO dataset was cleaned by removing duplication and redundancy, because both will introduce bias to the downstream comparisons and evaluations (See Figure \ref{fig:methods}). The clean maize GO datasets were combined to generate an aggregate maize-GAMER dataset. The aggregation of component datasets introduced additional duplication and redundancy, so the aggregate dataset was also cleaned by removing duplication and redundancy. 

%<DUPLICATE SECTIONS - my edits are in the methods - CMA>
%\subsection{%Cleaning \& Combining GO Annotation Method datasets}

%\subsubsection{%Removing Redundancy and Duplication}
%Closer inspection of the annotation datasets from different GO annotation methods revealed two different factors that were influencing the variation of GO annotations, namely duplication and redundancy. Duplication is defined as the same gene annotated with the same GO term multiple times in a single annotation dataset. This clearly inflates the number of annotations, and could potentially bias evaluations in GO annotation datasets, especially in computational annotation datasets.

%The GO terms are hierarchically related to other terms on a directed acyclic graph. This enables ancestral terms to be inherited from descendant terms. Redundancy is when a gene is annotated to a descendent term and is also annotated to one or more ancestral terms  of the specific descendant term. Redundancy also inflates the number of annotations in the dataset and could potentially bias the following evaluations.

%Duplication and Redundancy are also seen in manually curated datasets. Duplication indicates different types of evidences supporting the same annotation, and redundancy indicates evidences that support different levels of GO hierarchy. This is not the case in high-throughput computational annotations, where all annotations will get the same evidence code (IEA). GO annotation dataset created by each method was cleaned by removing redundant and duplicate annotations. While all methods had at least some redundancy or duplication, the numbers varied from method to method (See table XXX). Higher duplication was seen in maize-UniProt-Plant dataset from sequence similarity method and maize-IPRS dataset from domain presence based method. Mixed-method pipelines had the highest redundancy than other methods. Mixed-method pipelines used in maize-GAMER have been designed to assign a score to all GO terms, and do not have steps to reduce redundancy when the results are reported.
%All datasets were cleaned by removing duplication and redundancy, and clean datasets were created for all 6 datasets from maize-GAMER. The number of annotations retained after cleaning was substantially lower for most datasets (see Table XXX). With the combined cleaning of both redundancy and duplication most datasets only retained less than half of the raw annotations. Only dataset that retained more than 50\% original after cleaning was the maize-TAIR dataset. All other datasets were either highly redundant or highly duplicated.

%\subsubsection{%maize-GAMER Aggregate Dataset}

%<Should we call these annotations rather than subsets for clarity - CMA>
%The clean datasets from GO annotation methods were combined together to create an aggregate dataset for maize-GAMER. The aggregate dataset is the union set of GO annotation datasets from different tools (i.e. Each annotation in the aggregate dataset was present in one or more GO annotation method datasets). Some annotations in the aggregate dataset were supported by multiple GO annotation method datasets, and this introduced duplication to the aggregate dataset. Similarly, some GO annotation methods assigned GO terms at different levels of GO hierarchy, and this introduced redundancy. The aggregate dataset was cleaned by removing duplication and redundancy.